src/libraries/System.Numerics.Tensors/tests/TensorPrimitivesTests.cs

```diff
@@ -0,0 +1,633 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+#pragma warning disable xUnit1025 // reporting duplicate test cases due to not distinguishing 0.0 from -0.0
+
+namespace System.Numerics.Tensors.Tests
+{
+    public static class TensorPrimitivesTests
+    {
+        private const int TensorSize = 512;
```

Comment:
Do we have any tests for boundary conditions, e.g. empty? Or error conditions?  Or is the plan for those to come separately?

Reply:
I added tests for the exception cases already. I can add ones for empty as well.

For floating-point correctness cases (i.e. `infinity`, `nan`, `-0`, etc), I think it'd be good to add handling for after we move off the "naive" implementation.